### DISCUSSION TITLE
How to disable automatic bold formatting of brackets

### DISCUSSION THREAD

**Discussion by svlandeg** [Textualize/rich#3364] [Q&A]
I have a description that I want to format with Rich markup, add extra information that should be escaped, and then I want to export this to HTML:

```
    from rich.console import Console
    from rich import print, markup

    description = "The name of the [green]user[/green]"
    extra = markup.escape("[required]")
    input_text = f"{description} {extra}"

    console = Console(record=True)
    with console.capture():
        console.print(input_text)

    html_output = console.export_html(inline_styles=True, code_format="{code}")
```
The `html_output` is 
```
The name of the <span style="color: #008000; text-decoration-color: #008000">user</span> <span style="font-weight: bold">[</span>required<span style="font-weight: bold">]</span>
```

Which mostly looks good, apart from the `[` and `]` getting `font-weight: bold`. Ideally, I'd like to obtain:

```
The name of the <span style="color: #008000; text-decoration-color: #008000">user</span> [required]
```

I was thinking that maybe `highlight=False` would help in the `Console`'s constructor, but if I do that then I loose the part entirely, and the output is
```
The name of the <span style="color: #008000; text-decoration-color: #008000">user</span> 
```

How can I include this `[required]` bit into the string and keep it as such through the `export_html` function?

**Comment by willmcgugan** [comment]
You can switch off both highlighting and markup by setting `markup=False, highlight=False` on the `print` call.

Alternatively, you can use [Text](https://rich.readthedocs.io/en/latest/text.html) objects which give you fine control over text style.

**Reply by svlandeg** [reply]
Thanks for the quick reply!

`markup=False` doesn't work here because no HTML tags are inserted if I use that.

I looked into this again and it appears that `console = Console(record=True, highlight=False)` is in fact the right answer (in combination with the `markup.escape` part I had in the original snippet) as it does in fact give me 

```
The name of the <span style="color: #008000; text-decoration-color: #008000">user</span> [required]
```

When printing this with `rich.print()`, the final part `[required]` is removed as it's unknown formatting. I think that's as expected though. Thanks again!
